src/components/InputPage.tsx

```diff
@@ -0,0 +1,92 @@
+import { useState, useEffect } from "react";
+import { entry, entryForDisplay } from "./interfaces";
+import { History } from "./History";
+import axios from "axios";
+
+function InputPage(): JSX.Element {
+    const [currentTitle, setCurrentTitle] = useState("");
+    const [currentText, setCurrentText] = useState("");
+    const [currentEntryToSend, setCurrentEntryToSend] = useState<entry>();
+    const [entriesFromApi, setEntriesFromApi] = useState<entryForDisplay[]>([]);
+
+    useEffect(() => {
+        function getEntries() {
+            axios
+                .get("https://paste-bin-si-tl.onrender.com/")
+                .then((response) => setEntriesFromApi(response.data))
+                .catch((error) => console.log(error));
+        }
+        getEntries();
+    }, []);
+
+    useEffect(() => {
+        function sendEntryToApi() {
+            axios
+                .post(
+                    "https://paste-bin-si-tl.onrender.com/",
+                    currentEntryToSend
+                )
+                .catch((error) => console.log(error))
+                .finally(() => getUpdatedEntries());
+        }
+
+        sendEntryToApi();
+    }, [currentEntryToSend]);
+
+    function getUpdatedEntries() {
+        axios
+            .get("https://paste-bin-si-tl.onrender.com/")
+            .then((response) => setEntriesFromApi(response.data))
+            .catch((error) => console.log(error));
+    }
+
+    function handleTitleInput(title: string) {
+        setCurrentTitle(title);
+    }
+
+    function handleText(text: string) {
+        setCurrentText(text);
+    }
+
+    function handleSubmit() {
+        const oneEntry: entry = {
+            title: currentTitle,
+            text: currentText,
+        };
+
+        if (currentText.trim().length > 0) {
+            setCurrentEntryToSend(oneEntry);
+        } else {
+            alert("Text body cannot be empty!");
+        }
+    }
+    const EntriesToDisplay: JSX.Element[] = [];
+    for (const element of entriesFromApi) {
+        EntriesToDisplay.push(<History oneEntry={element} />);
+    }
```

Comment:
because the get request fetches all table entries you might be processing too much data unnecessarily - for instance if there were 1000s of entries this for loop makes an array of all of them when you only really want 10.